src/js/shared/Http/index.mjs

```diff
@@ -0,0 +1,123 @@
+let baseUri
+let tokenApiResolver
+let tokenApiReady = new Promise( resolve => tokenApiResolver = resolve )
+
+let authApiResolver
+let authApiReady = new Promise( resolve => authApiResolver = resolve )
+
+
+let proxyProvider = (resource, options) => Promise.resolve({ resource, options })
+
+const join = (...args) => args.map(a => (''+a).split('/')).flat().filter(a => a!=='').join('/').replace(':/', '://')
+
+function setEndpoint(endpoint) {
+    baseUri = endpoint
+}
+
+function onToken(callback) {
+    if (tokenApiResolver) {
+        tokenApiResolver(callback)
+        tokenApiResolver = null
+    }
+}
+
+function getToken() {
+    return tokenApiReady.then(tokenGetter => tokenGetter())
+}
+
+function onAuthorize(callback) {
+    if (authApiResolver) {
+        authApiResolver(callback)
+        authApiResolver = null
+    }
+}
+
+function getAuthorization(grants) {
+    return authApiReady.then(authGetter => authGetter(grants))
+}
+
+
+// this gets called syncrhonously by the SDK itself, if needed
+export function proxy(callback) {
+    proxyProvider = callback
+}
+
+function send(module, method, params, http = {}) {
+
+    let path = http.path || `/${module}/${method}`
+    const headers = http.headers || {}
+    const body = params && JSON.stringify(params) || null
+    let query = http.parameters || ''
+    
+    Object.entries(params).forEach(([name, value]) => {
+        const find = '${param.' + name + '}'
+        path = path.replace(find, value)
```

Comment:
This will only replace a single occurrence, guess that is probably the usual case, but might be better to do a replace all?
However, I found this out the hard way, WPE does not support replaceAll function.
You have to do this:
```
const re = new RegExp('\\${param.' + name + '}', 'g')
path = path.replace(re, value)
```

Reply:
Thanks for the heads up. we should file a bug with WPE, too.